File: Software/bootloader-icobs-light-project/src/crc8.c
/**
 * author: Guillaume Patrigeon
 * update: 11-02-2018
 */

#include "crc8.h"

#define CRCPOL 7

unsigned char CRC8_TABLE[256];

void CRC8_Init(void)
{
	int i, j;
	unsigned int k;

	for (i=0; i<256; i++)
	{
		k = i;

		for (j=0; j<8; j++)
		{
			k <<= 1;
			if (k & 0x100)
				k ^= CRCPOL;
		}

		CRC8_TABLE[i] = k;
	}
}

File: GameIcobs/Software/Codeloader_ubuntu/lib_loader/crc8.py
#!/usr/bin/env python3
# author: Guillaume Patrigeon
# update: 23-03-2017



class CRC8():
	CRC = 0

	def __init__(self, polynomial=7):
		self.CRC = 0
		self.table = []
		self.setPolynomial(polynomial)


	def setPolynomial(self, polynomial):
		self.polynomial = polynomial
		del self.table[:]

		for i in range(256):
			k = i

			for j in range(8):
				k <<= 1
				if k & 0x100:
					k ^= self.polynomial

			self.table.append(k & 0xFF)


	def pushByte(self, b):
		self.CRC = self.table[(self.CRC ^ b) & 0xFF]



if __name__ == "__main__":
	crc = CRC8()

	crc.pushByte(0xAB)
	crc.pushByte(0xCD)

	print("%02X" % crc.CRC)
